### PR TITLE
Integrations are now Apps!

### PR DESCRIPTION
**Pull request by m1guelpf**
See #591

### CODE REVIEW

File: lib/Github/Api/Apps.php [hunk 1]
```diff
@@ -5,28 +5,14 @@
...
     use AcceptHeaderTrait;
 
     /**
-     * Configure the accept header for Early Access to the integrations api
-     *
-     * @see https://developer.github.com/v3/integrations/
-     *
-     * @return self
-     */
-    public function configure()
```
Comment (acrobat) [inline lib/Github/Api/Apps.php]: I think we should keep BC here and just remove the `$this->acceptHeaderValue = 'application/vnd.github.machine-man-preview+json';` but keep the method as users of the lib are possibly calling this method

File: lib/Github/Api/Apps.php [hunk 2]
```diff
@@ -5,28 +5,14 @@
 use Github\Api\AcceptHeaderTrait;
 
 /**
- * @link   https://developer.github.com/v3/integrations/
+ * @link   https://developer.github.com/v3/apps/
  * @author Nils Adermann <naderman@naderman.de>
  */
-class Integrations extends AbstractApi
+class Apps extends AbstractApi
```
Comment (acrobat) [inline lib/Github/Api/Apps.php]: Also to keep BC the class can not be renamed (in my opinion). Bu the class can be deprecated and a new Apps class can be created (where this class can extend from)

Comment (m1guelpf) [inline lib/Github/Api/Apps.php]: As I also modified the Client to point to the new class, there's no reason to keep using this...

Comment (acrobat) [inline lib/Github/Api/Apps.php]: I know, I wanted to do a same sort of change but @Nyholm came up with a good point.

**Comment by m1guelpf** [comment]
@acrobat Done!

**Comment by m1guelpf** [comment]
@acrobat @Nyholm Done!

File: lib/Github/Api/Apps.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+namespace Github\Api;
+
+/**
+ * @link   https://developer.github.com/v3/apps/
+ * @author Nils Adermann <naderman@naderman.de>
+ */
+class Apps extends AbstractApi
+{
+    /**
+     * @deprecated
```
Comment (Nyholm) [inline lib/Github/Api/Apps.php]: Why include this?

File: lib/Github/Api/Integrations.php [hunk 4]
```diff
@@ -2,107 +2,11 @@
 
 namespace Github\Api;
 
-use Github\Api\AcceptHeaderTrait;
-
 /**
- * @link   https://developer.github.com/v3/integrations/
+ * @deprecated
```
Comment (Nyholm) [inline lib/Github/Api/Integrations.php]: Add a comment saying what the new thing is.

File: lib/Github/Api/Apps.php (hunk 3 above)
Comment (Nyholm) [inline lib/Github/Api/Apps.php]: Oh I see. Move this method back to the Integrations class. It is weird to add a class with deprecated functions.

File: lib/Github/Client.php [hunk 5]
```diff
@@ -29,8 +29,9 @@
  * @method Api\Gists gist()
  * @method Api\Gists gists()
  * @method Api\Miscellaneous\Gitignore gitignore()
- * @method Api\Integrations integration()
```
Comment (Nyholm) [inline lib/Github/Client.php]: You could leave these comments.

File: lib/Github/Client.php [hunk 6]
```diff
@@ -200,7 +201,8 @@ public function api($name)
 
             case 'integration':
             case 'integrations':
-                $api = new Api\Integrations($this);
```
Comment (Nyholm) [inline lib/Github/Client.php]: You should keep this. Calling integrations should give you the deprecated Integrations class, right?

Comment (acrobat) [inline lib/Github/Client.php]: That's right, this way the user knows he has to update some code

File: lib/Github/Api/Integrations.php (hunk 4 above)
Comment (acrobat) [inline lib/Github/Api/Integrations.php]: Maybe we can do it "symfony" style with a `@trigger_error` when using the class. Like this https://github.com/symfony/symfony/blob/3.3/src/Symfony/Component/ClassLoader/ApcClassLoader.php#L14

But also keep the annotation so ide's will show it. What do you think @Nyholm?

File: lib/Github/Client.php (hunk 5 above)
Comment (acrobat) [inline lib/Github/Client.php]: yes indeed keep the old methods (to the deprecated class) so a user will know they have to change their code

Comment (m1guelpf) [inline lib/Github/Client.php]: Done!

File: lib/Github/Client.php [hunk 7]
```diff
@@ -201,6 +201,8 @@ public function api($name)
 
             case 'integration':
             case 'integrations':
+                $api = new Api\Integrations($this);
+                break
```
Comment (acrobat) [inline lib/Github/Client.php]: missing `;` here

File: lib/Github/Client.php (hunk 5 above)
Comment (acrobat) [inline lib/Github/Client.php]: :+1:

**Comment by acrobat** [comment]
Yeah it looks like packagist is also having problems https://twitter.com/packagist/status/866952618349920257

**Comment by m1guelpf** [comment]
@acrobat @Nyholm Could you retry the build?

**Comment by acrobat** [comment]
@m1guelpf I've restarted the build!

**Comment by acrobat** [comment]
The build is green again! Thanks @m1guelpf

**Pull request merged**